### PR TITLE
#165593726 Implement getByEmail endpoint 

### PR DESCRIPTION
**Pull request by okabamac**
## What does this PR do?
Implement the getByEmail endpoint

Remove email from token

Change validation error status code

Move validation middleware to routes

## Description of Task to be completed?

have the following endpoint working
GET /api/v1/users/:email/accounts

# How should this be manually tested?
clone the repo git clone git@github.com:okabamac/banka.git

on your terminal run cd journal

run git checkout ft-165582997

run npm install

run npm run dev

make a GET request to the route

Any background context you want to provide?

NA

## What are the relevant pivotal tracker stories?

[#165593726](https://www.pivotaltracker.com/story/show/165582997)
[#165601668](https://www.pivotaltracker.com/story/show/165601668)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 node_modules
-.env
\ No newline at end of file
+.env
+.vscode
```
Comment (Mrshinezee) [inline .gitignore]: Add new line

File: src/controllers/accountController.js [hunk 2]
```diff
@@ -77,17 +77,18 @@ class AccountControl {
 
     if (validCreateAccount.statusCode === 422) return;
```
Comment (Mrshinezee) [inline src/controllers/accountController.js]: Why are you checking status code.

File: src/controllers/userController.js [hunk 3]
```diff
@@ -39,6 +39,31 @@ class UserControl {
     }
   }
 
+  static async getAllByEmail(req, res, next) {
+    const {
+      email,
+    } = req.params;
+
+    if (email == req.decoded.email) {
```
Comment (Mrshinezee) [inline src/controllers/userController.js]: Remove email from token

File: .gitignore (hunk 1 above)
Comment (okabamac) [reply .gitignore]: OK

File: src/controllers/accountController.js (hunk 2 above)
Comment (okabamac) [reply src/controllers/accountController.js]: The middleware I am using returns unhandled promise if I dont

File: src/controllers/userController.js (hunk 3 above)
Comment (okabamac) [reply src/controllers/userController.js]: These are the things I will be working on the feedback

**Comment by Mrshinezee** [comment]
ft(getByEmail)-165582997-user-can-view-by-email - Check the documentation on how it should look.

**Pull request merged**